### PR TITLE
Enum api

### PR DESCRIPTION
**Pull request by RyadaProductions**
This should finish up the majority of all the enumization throughout the system. In addition to the usage of `[Display()]` so that other developers can still get a UI friendly name.

### CODE REVIEW

**Comment by RyadaProductions** [comment]
@MingweiSamuel would you be able to take a look at this PR?
I opted to leave the enums as is, and maybe change them later on. As i would like to have some real life tests going soon on the Enum api.

File: Camille/gen/dotUtils.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ function normalizePropName(propName, schemaName, value) {
   return name;
 }
 
-function stringifyType(prop, endpoint = null, nullable = false) {
+function stringifyType(prop, name = '', endpoint = null, nullable = false) {
```
Comment (MingweiSamuel) [inline Camille/gen/dotUtils.js]: Appears `name` param is no longer needed because of `x-enum`?

File: Camille/gen/Enums/Champion.cs.dt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        .sort(({ name: a }, { name: b }) => a > b ? 1 : -1);
+    const hashFactor = 256;
+    const enumName = function(name) { return name.toUpperCase().replace(/[^A-Z]+/, '_'); };
+    const strHash = function(str) {
+        let h = 0;
+        for (let c of str)
+            h = hashFactor * h + c.charCodeAt(0);
+        return h;
+    };
+    const padId = function(id) { return ('' + id).padEnd(3); };
```
Comment (MingweiSamuel) [inline Camille/gen/Enums/Champion.cs.dt]: We can remove these and hashFactor above unless you have a plan for them

(We will need a way to get the champion "key" from the champion enum though, since ddragon tends to use those over the ids. (i.e. "Wu Kong" is `MonkeyKing`), but that can be a separate PR)

**Review by MingweiSamuel — COMMENTED** [review]
Sorry, lost track of this one.

I'm not sure about the `[System.Text.Json.Serialization.JsonConverter(typeof(System.Text.Json.Serialization.JsonStringEnumConverter))]` attributes since most of the enums are represented using their number. How do they work?

Edit it seems to work with and without it on core 3.1?

File: Camille/src/Enums/Tier.cs [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
+        DIAMOND,
+        PLATINUM,
+        GOLD,
+        SILVER,
+        BRONZE,
+        IRON,
+
+        /// <summary>In most endpoints, tier will not be provided if related summoner is unranked.</summary>
+        UNRANKED,
+    }
```
Comment (MingweiSamuel) [inline Camille/src/Enums/Tier.cs]: This is from the previous commits, but in the current version `0` or missing values will get parsed as Challenger. API seems to have stopped returning `highestAchievedSeasonTier` so that is one place where that is noticeable.

This will be fixed with proper DTO field nullability which I've already annotated riotapi-schema with. However we should probably start the non-numeric enums from 1 to catch any future occurrences.

Comment (RyadaProductions) [reply Camille/src/Enums/Tier.cs]: I am not sure how the returning values work, wouldn't it in this case make sense to put UNRANKED as the `0` in the enum?

File: Camille/gen/Enums/Champion.cs.dt (hunk 2 above)
Comment (RyadaProductions) [reply Camille/gen/Enums/Champion.cs.dt]: I will remove these functions and see if I can find a graceful solution for the champion key.

File: Camille/gen/dotUtils.js (hunk 1 above)
Comment (RyadaProductions) [reply Camille/gen/dotUtils.js]: Forgot about that one, will remove it

**Comment by RyadaProductions** [comment]
The `[System.Text.Json.Serialization.JsonConverter(typeof(System.Text.Json.Serialization.JsonStringEnumConverter))]` is necessary in 3.0 because somehow when deserializing the objects it can't properly convert a string to enum value. This helps getting around it, and makes us able to use it in 3.0, aside from the weird bug with deserializing

File: Camille/src/Enums/Tier.cs (hunk 3 above)
Comment (MingweiSamuel) [reply Camille/src/Enums/Tier.cs]: UNRANKED wouldn't be the same as no value, it seems highestAchievedSeasonTier is straight-up removed https://github.com/RiotGames/developer-relations/issues/94

I'm not worried about it too much, just something I took note of
